src/Voron/Impl/Paging/Windows32BitsMemoryMapPager.cs

```diff
@@ -241,12 +241,15 @@ public override bool ShouldLockMemoryAtPagerLevel()
         private void LockMemory32Bits(byte* address, long sizeToLock, TransactionState state)
         {
             try
-            {
+            {                
                 if (Sodium.sodium_mlock(address, (UIntPtr)sizeToLock) != 0)
                 {
-                    if (DoNotConsiderMemoryLockFailureAsCatastrophicError == false)
-                    {
-                        TryHandleFailureToLockMemory(address, sizeToLock, state.TotalLoadedSize * 2);
+                    lock (WorkingSetIncreaseLocker)
+                    { 
+                        if (Sodium.sodium_mlock(address, (UIntPtr)sizeToLock) != 0 && DoNotConsiderMemoryLockFailureAsCatastrophicError == false)
```

Comment:
Why is the `DoNotConsiderMemoryLockFailureAsCatastrophicError ` check inside the lock?

Reply:
This is inside lock because it applies to the result of `sodium_mlock` from the `if` here

Reply:
done

Reply:
okay